### PR TITLE
feat: email cron job

### PR DESCRIPTION
**Pull request by akshay2000saxena**
## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Email-Reminders-Cron-Job-9d8cdba9a68141728d5696759d3f7c06)

## Implementation description
* Added a cron job which sends emails

## Steps to test
1. Adjust values in implementation/emailService.ts line 29 to '* * * * *'
2. Add values to the scheduling table
3. Check if emails were sent

## What should reviewers focus on?
* 

## Checklist
- [X] My PR name is descriptive and in imperative tense
- [X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/package.json [hunk 1]
```diff
@@ -68,8 +68,10 @@
     "eslint-plugin-import": "^2.22.1",
     "eslint-plugin-prettier": "^3.3.1",
     "jest": "^27.0.4",
+    "moment": "^2.20.1",
```
Comment (hanlinc27) [inline backend/typescript/package.json:71]: I believe `moment` is deprecated, we [could use](https://momentjs.com/docs/#/-project-status/recommendations/) `Day.js` or `date-fns` ?

Comment (akshay2000saxena) [reply backend/typescript/package.json]: updated to use dayjs

**Review by Shehryar21 — APPROVED** [review]

File: backend/typescript/services/implementations/emailService.ts [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+import dayjs from "dayjs";
 import IEmailService from "../interfaces/emailService";
 import { NodemailerConfig } from "../../types";
 import logger from "../../utilities/logger";
+import Schedule from "../../models/scheduling.model";
+import User from "../../models/user.model";
+import createReminderEmailContent from "../../utilities/emailUtils";
+
+// eslint-disable-next-line
+const cron = require("node-cron");
```
Comment (hanlinc27) [inline backend/typescript/services/implementations/emailService.ts:11]:
```suggestion
import * as cron from 'node-cron';
```
will this work? if not then just leave as is

might need `yarn add @types/node-cron`

File: backend/typescript/utilities/emailUtils.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  // TODO: need a final email body from designers
+
+  return `
+      <html>
+        <body>
+          Test content: ${user.email}
+        </body>
+      </html>
+    `;
+};
```
Comment (hanlinc27) [inline backend/typescript/utilities/emailUtils.ts:14]:
```typescript suggestion
const createReminderEmailContent = (schedule: Schedule, user: User): string => (`
      <html>
        <body>
          Test content: ${user.email}
          Schedule: ${scheudule} 
        </body>
      </html>
    `);
```

**Review by hanlinc27 — APPROVED** [review]

**Pull request merged**